packages/authorize/src/useAuthorize.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from 'react-query';
+
+import { checkPermissions } from './api';
+import type { QueryOptions, RequestedPermissions, RequestedResources } from './types';
+
+type Parameters = {
+  organizationId?: string;
+  customerId?: string;
+  region?: boolean | string;
+  resources?: RequestedResources;
+};
+
+type Result = { authorized: boolean; isLoading: boolean };
+
+const useAuthorize = (
+  permissions: RequestedPermissions,
+  parameters: Parameters = {},
+  options?: QueryOptions
+): Result => {
+  const { organizationId, customerId, region = true, resources } = parameters;
+
+  const { data: authorized = false, isLoading } = useQuery(
+    ['authorized', permissions, region, resources, organizationId, customerId],
```

Comment:
May want to change the first part of this key to `'useAuthorize'` instead of `'authorized'`. IMO it would make it clearer where things came from if you are using something like `queryClient.getQueryData(queryKey)` in another part of the codebase